server/app/mutations/domain_authorizations/authorize.rb

```diff
@@ -23,11 +23,16 @@ def validate
       unless grid.grid_secrets.find_by(name: LE_PRIVATE_KEY)
         add_error(:le_registration, :missing, "Let's Encrypt registration missing")
       end
-      if self.authorization_type == 'tls-sni-01'
-        add_error(:linked_service, :missing, "Service link needs to be given for tls-sni-01 authorization type") unless self.linked_service
+      if self.authorization_type == 'tls-sni-01' && self.linked_service.nil?
+        add_error(:linked_service, :missing, "Service link needs to be given for tls-sni-01 authorization type")
       end
       if self.linked_service
-        add_error(:linked_service, :not_found, "Linked service needs to point to existing service") unless @linked_service = resolve_service(self.grid, linked_service)
+        @lb_service = resolve_service(self.grid, linked_service)
+        if @lb_service.nil?
+          add_error(:linked_service, :not_found, "Linked service needs to point to existing service")
+        elsif @lb_service && @lb_service.net != 'host' && !@lb_service.ports.any? { |p| p['node_port'] == 443 }
```

Comment:
would it makes sense to have this kind of check on the `GridService` model? I mean something like `@lb_service.https_open?`

Reply:
Imho does not make really sense to move this logic into the `GridService` model. I can refactor this code if necessary ...

Reply:
for me having many conditionals and some of those negated, makes it hard to read. :) maybe simplify it by just removing the first `@lb_service` "condition", it shouldn't be falsey in the `elsif` branch?

Reply:
Refactored.